ipykernel/comm/manager.py

```diff
@@ -91,13 +62,12 @@ def get_comm(self, comm_id):
         This will not raise an error,
         it will log messages if the comm cannot be found.
         """
-        if comm_id not in self.comms:
+        try:
+            return self.comms[comm_id]
+        except KeyError:
             self.log.warn("No such comm: %s", comm_id)
-            self.log.debug("Current comms: %s", lazy_keys(self.comms))
-            return
-        # call, because we store weakrefs
-        comm = self.comms[comm_id]
-        return comm
+            if self.log.isEnabledFor(self.log.DEBUG):
+                self.log.debug("Current comms: %s", self.comms.keys())
```

Comment:
Needs `list(self.comms.keys())` in order to get the actual keys object, rather than the `dict_keys()` generator.